File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Call/Views/ActionsMenuView/MenuAction/MenuAction.m
//
//  MenuAction.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 08.10.2020.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "MenuAction.h"

@interface MenuAction ()
//MARK: - Properties
@property (strong, nonatomic) NSString *title;
@property (assign, nonatomic) UserAction action;
@property (assign, nonatomic) BOOL isSelected;
@property (strong, nonatomic) MenuActionHandler handler;

@end

@implementation MenuAction
//MARK: - Life Cycle
- (instancetype)initWithTitle:(NSString *)title isSelected:(BOOL)isSelected action:(UserAction)action handler:(MenuActionHandler _Nullable)handler {
    self = [super init];
    if (self) {
        self.title = title;
        self.action = action;
        self.isSelected = isSelected;
        self.handler = handler;
    }
    return self;
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/ActionsMenuViewController/MenuAction.swift
//
//  MenuAction.swift
//  sample-chat-swift
//
//  Created by Injoit on 13.08.2020.
//  Copyright © 2020 quickBlox. All rights reserved.
//

import Foundation

typealias SelectedAction = (() -> Void)

class MenuAction {
    //MARK: - Properties
    var successHandler: SelectedAction?
    var title: String
    
    //MARK: - Life Cycle
    init (title: String,
          handler: @escaping SelectedAction) {
        self.title = title
        self.successHandler = handler
    }
}
